tests/benchmark/bm_utils.h

```diff
@@ -7,58 +7,64 @@
 #include <benchmark/benchmark.h>
 #include <random>
 #include <unistd.h>
+#include <istream>
 #include "VecSim/vec_sim.h"
 #include "VecSim/vec_sim_interface.h"
 #include "VecSim/query_results.h"
 #include "VecSim/utils/arr_cpp.h"
 #include "VecSim/algorithms/brute_force/brute_force.h"
-#include "VecSim/algorithms/hnsw/hnsw_single.h"
+#include "VecSim/algorithms/hnsw/hnsw.h"
+#include "VecSim/algorithms/hnsw/hnsw_factory.h"
+#include "bm_definitions.h"
 
-class BM_VecSimBasics : public benchmark::Fixture {
+// This class includes every static data member that is:
+// 1. common for fp32 and fp64 data sets.
+// or 2. in use for all types of bench mark, if this type
+// is defined in a separate compilation unit.
+class BM_VecSimGeneral : public benchmark::Fixture {
 public:
-    static VecSimIndex *bf_index;
-    static VecSimIndex *hnsw_index;
+    static size_t block_size;
```

Comment:
any reason for this is the only public member?

Reply:
yes, used to define the number of iterations